src/server.py

```diff
@@ -27,38 +31,53 @@ def server():
             if len(part) < buffer_length:
                 message_complete = True
         print(message)
-        conn.send(response_ok())
+        try:
+            parse_request(message)
+        except HTTPException as exception:
+            conn.send((response_error(exception).encode("utf-8")))
+            conn.close()
+        else:
+            conn.send(response_ok())
         conn.close()
 
 
+def parse_request(request):
+    """Parse incoming http request for errors and return uri."""
+    request2 = request.decode('utf8')
+    split_msg = request2.split(CRLF + CRLF, 1)
+    head = split_msg[0]
+    head_lines = head.split(CRLF)
+    first_line = head_lines.pop(0)
+    try:
+        method, path, proto = first_line.split()
```

Comment:
I would not call this "path" yet.  It is a URI, not a path.  It will become a path when we use it to find a filesystem resource, but for now you don't want to confuse yourselves into thinking that it is a path here.